packages/e2e-tests/scripts/save-examples.ts

```diff
@@ -242,28 +243,44 @@ async function saveBrowserExample({ example }: TestRunCallbackArgs) {
     console.log("Example finished");
   }
   const playwrightScript: PlaywrightScript = example.playwrightScript ?? defaultPlaywrightScript;
-  await raceForTime(
-    CONFIG.recordingTimeout,
-    recordPlaywright(async (page, expect) => {
-      const waitForLogPromise = playwrightScript(page, expect);
-      const goToPagePromise = page.goto(exampleUrl);
-
-      await Promise.all([goToPagePromise, waitForLogPromise]);
-    })
-  );
 
-  const recordingId = await raceForTime(CONFIG.uploadTimeout, uploadLastRecording(exampleUrl));
-  if (recordingId == null) {
-    throw new Error(`Recording "${example.filename}" not uploaded`);
-  }
-  exampleToNewRecordingId[example.filename] = recordingId;
+  let recordingError: any;
+  for (let i = 0; i < CONFIG.recordingAttempts; ++i) {
+    try {
+      await raceForTime(
+        CONFIG.recordingTimeout,
+        recordPlaywright(async (page, expect) => {
+          const waitForLogPromise = playwrightScript(page, expect);
+          const goToPagePromise = page.goto(exampleUrl);
+
+          await Promise.all([goToPagePromise, waitForLogPromise]);
+        })
+      );
+
+      const recordingId = await raceForTime(CONFIG.uploadTimeout, uploadLastRecording(exampleUrl));
+      if (!recordingId) {
+        throw new Error(`Recording "${example.filename}" not uploaded`);
+      }
+      exampleToNewRecordingId[example.filename] = recordingId;
+
+      if (config.useExampleFile && recordingId) {
+        await saveRecording(example.filename, config.replayApiKey, recordingId, true);
+      }
 
-  if (config.useExampleFile && recordingId) {
-    await saveRecording(example.filename, config.replayApiKey, recordingId, true);
+      if (recordingId) {
+        removeRecording(recordingId);
+      }
+      break;
```

Comment:
What happens if recording fails the first time but then succeeds on retry? Seems like this would still throw and say that recording failed.

Am I reading it wrong? Seems like we would want a retry to succeed.

Seems like the most straight forward way to do that would be to return here rather than _breaking_.
```suggestion
      return;
```

Reply:
I don't want a retry to succeed. The root cause here is likely/often going to be a recorder crash, and we want to show no leniency toward those! I am only retrying to get a verbose error output 🙄

Reply:
Ah, I see. In that case, I think the "recordingAttempts" variable is kind of misleading (at least to me). You're not really _attempting_ some number of times to get a usable recording. You're trying once, and if it fails, you're re-running in verbose mode to get more error info. Seems like that may be more clear if you just hard-coded the retry logic and used a boolean (e.g. `retryingWithVerboseMode`) to track the current case.